x/delegation/keeper/un_delegation_state.go

```
@@ -188,33 +220,114 @@ func (k *Keeper) IterateUndelegationsByStakerAndAsset(
+		pendingUndelegationKeys, err := types.ParsePendingUndelegationKey(iterator.Key())
+		if err != nil {
+			return err
+		}
+		if isCompletable {
+			// Due to the current implementation, the completion of undelegation is triggered
+			// by per-block checks rather than using epoch hooks. As a result, when the epoch
+			// number to be completed equals the current epoch number, the undelegation remains
+			// in a pending state. It will only be completed after the current epoch ends, specifically
+			// in the first block of the next epoch.
+			// This logic might be changed if we chose epochHook to complete the undelgation in the future.
+			if pendingUndelegationKeys.EpochNumber >= uint64(currentEpoch) {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int64 -> uint64

(gosec)